trunner/device.py

```diff
@@ -30,13 +33,203 @@
 }
 
 
-def proccess_log_output(proc):
-    while True:
-        output = proc.stdout.readline().decode('utf-8')
-        if proc.poll() is not None and output == '':
-            break
-        if output:
-            logging.info(output)
+def is_github_actions():
+    return os.getenv('GITHUB_ACTIONS', False)
+
+
+class Psu:
+    """Wrapper for psu program"""
+
+    def __init__(self, script, cwd=_BOOT_DIR):
+        self.script = script
+        self.cwd = cwd
+        self.proc = None
+
+    def read_output(self):
+        if is_github_actions():
+            logging.info('::group::psu\n')
```

Comment:
Unless intention is to output empty line trailing newlines should be removed from logging functions. There are more cases in this commit.

Reply:
Is a more complex problem right now. Logging doesn't provide `end` arguments as print, so I set logging to not use a new line by default. I'm going to change it and provide more coherent way of logging. So I suggest leaving it as it is right now and fix it for the whole runner in a separate PR.